src/librustdoc/core.rs

```diff
@@ -416,10 +447,16 @@ pub fn run_core(options: RustdocOptions) -> (clean::Crate, RenderInfo, RenderOpt
             let mut global_ctxt = abort_on_err(queries.global_ctxt(), sess).take();
 
             global_ctxt.enter(|tcx| {
-                tcx.analysis(LOCAL_CRATE).ok();
-
-                // Abort if there were any errors so far
-                sess.abort_if_errors();
+                // Some queries require that they only run on valid types:
+                // https://github.com/rust-lang/rust/pull/73566#issuecomment-656954425
+                // Therefore typecheck this crate before running lints.
+                // NOTE: this does not typeck item bodies or run the default rustc lints
+                // (see `override_queries` in the `config`)
```

Comment:
`rustc_typeck::check_crate` would not run lints, so that part isn't relevant. Although you could say that you don't call `tcx.analysis()` in order to not run lints and other checks.